### PR TITLE
Watch all sources in config

### PR DESCRIPTION
**Pull request by Cactucs**

### CODE REVIEW

File: lib/tasks/watch_sources.js [hunk 1]
```diff
@@ -27,7 +29,7 @@ module.exports = function(gulp, config, watch) {
 		var minimatch = require('minimatch')
 		var path = require('path')
 		var runcmd = require('../helpers/runcmd')
-		var glob = '**/*.{css,styl,scss,less,js,jsx,es6,es,coffee,jade,json,html,htm}'
+		var glob = '**/*{' + fileExtensions.join() + "}"
```
Comment (enzy) [inline lib/tasks/watch_sources.js]: Please make it as extension of hardcoded types. There is some edge (eg. patternlab) cases which your change brakes.

Comment (Cactucs) [inline lib/tasks/watch_sources.js]: "extension of hardcoded types" means that it would watch only file extension in default mapping?

Comment (enzy) [inline lib/tasks/watch_sources.js]: I mean:  `'**/*.{css,styl,scss,less,js,jsx,es6,es,coffee,jade,json,html,htm, otherExtensions...}'`

Comment (Cactucs) [inline lib/tasks/watch_sources.js]: Fixed

File: lib/tasks/watch_sources.js [hunk 2]
```diff
@@ -13,12 +13,15 @@ module.exports = function(gulp, config, watch) {
...
 
+	var fileExtensions = []
 	Object.keys(mapping).forEach(function(key) {
 		mapping[key] = mapping[key].map(function(i) {
 			var ext = '.' + i.replace(/^\./, '') // remove . from the beginning of the string
+			fileExtensions.push(ext)
 			return new RegExp(ext + '$')
 		})
 	})
+	fileExtensions.push('css','styl','scss','less','js','jsx','es6','es','coffee','jade','json','html','htm')
```
Comment (enzy) [inline lib/tasks/watch_sources.js]: this should be on the line 16 inside a declaration

File: lib/tasks/watch_sources.js (hunk 1 above)
Comment (enzy) [inline lib/tasks/watch_sources.js]: What about duplicates? It's a good idea to filter them first.

File: lib/tasks/watch_sources.js (hunk 2 above)
Comment (Cactucs) [inline lib/tasks/watch_sources.js]: fixed!

File: lib/tasks/watch_sources.js (hunk 1 above)
Comment (Cactucs) [inline lib/tasks/watch_sources.js]: Good point. Fixed!

**Comment by enzy** [comment]
:+1:

**Pull request merged**